### PR TITLE
Update dependencies and cleanup code

### PR DESCRIPTION
**Pull request by ballinette**
Some changes to improve the code quality:
* update dependencies
* replace istanbul by nyc for code coverage
* use `for ... of` instead of `forEach(callback)`
* Now reject the plugin init if at least one probe is misconfigured

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -169,13 +171,13 @@ class ProbePlugin {
       event = request.input.body.event;
     debug(`Received measure for monitor ${event}`);
 
-    this.eventMapping.monitor[event].forEach(probe => {
+    for (const probe of this.eventMapping.monitor[event]) {
```
Comment (Aschen) [inline lib/index.js]: Why use iterator over forEach ?

Comment (ballinette) [reply lib/index.js]: to be consistent with the code of kuzzle core.

* it is easier to read
* it is faster

File: lib/index.js [hunk 2]
```diff
@@ -440,613 +442,602 @@ class ProbePlugin {
...
-  return promise;
-}
+  /**
+   * Performs a set of operations after receiving a probe.
+   *
+   * @param {Object} probe
+   * @param {Object} measure
+   */
+  _afterMeasure(probe, measure) {
+    return new Promise((resolve, reject) => {
```
Comment (scottinet) [inline lib/index.js]: should be `return new Bluebird((resolve, reject) => {`

Comment (ballinette) [reply lib/index.js]: indeed
Done.

**Pull request merged**